vault-plugins/flant_iam/model/roleresolver.go

```diff
@@ -0,0 +1,24 @@
+package model
+
+type RoleResolver interface {
+	IsUserSharedWith(TenantUUID) (bool, error)
+	IsServiceAccountSharedWith(TenantUUID) (bool, error)
+
+	CheckUserForProjectScopedRole(UserUUID, RoleName, TenantUUID, ProjectUUID) (bool, RoleBindingParams, error)
+	CheckUserForTenantScopedRole(UserUUID, RoleName, TenantUUID) (bool, RoleBindingParams, error)
+	CheckServiceAccountForProjectScopedRole(ServiceAccountUUID, RoleName, TenantUUID, ProjectUUID) (bool, RoleBindingParams, error)
+	CheckServiceAccountForTenantScopedRole(ServiceAccountUUID, RoleName, TenantUUID) (bool, RoleBindingParams, error)
+
+	FindSubjectsWithProjectScopedRole(RoleName, TenantUUID, ProjectUUID) ([]UserUUID, []ServiceAccountUUID, error)
+	FindSubjectsWithTenantScopedRole(RoleName, TenantUUID) ([]UserUUID, []ServiceAccountUUID, error)
+
+	CheckGroupForRole(GroupUUID, RoleName) (bool, error)
+}
+
```

Comment:
It's just a thought.

We have three (or just two, IDK) things to check this way: users, serviceaccount (and groups, maybe).

Can we use single narrow interface for them, in order not to widen the interface surface? I mean kind of this thing:

```go
type RoleResolver interface {
	IsSharedWithTenant(TenantUUID) (bool, error)

	HasProjectScopedRole(UserUUID, RoleName, TenantUUID, ProjectUUID) (bool, RoleBindingParams, error)
	HasTenantScopedRole(UserUUID, RoleName, TenantUUID) (bool, RoleBindingParams, error)

	FindInProjectScopedRole(RoleName, TenantUUID, ProjectUUID) ([]UUID, error)
	FindInTenantScopedRole(RoleName, TenantUUID) ([]UUID, error)
}

// these two implement RoleResolver
type UserRoleResolver struct {}                    
type ServiceAccountRoleResolver struct {}

// (maybe) implements RoleResolver
type GroupRoleResolver struct {}                
```